network/config.go

```diff
@@ -109,15 +107,10 @@ func (c *Config) build() error {
 }
 
 func (c *Config) checkState() (err error) {
-	if c.BlockState == nil {
+	if c.BlockState == nil && !c.NoStatus {
```

Comment:
Why was `&& !c.NoStatus` added here?

Reply:
it's an easy way to configure whether you need a BlockState or not in the tests

Reply:
maybe a comment to explain that, otherwise it looks a bit out of place

Reply:
sure thing